@xen-orchestra/web-core/lib/components/StackedBar.vue

```diff
@@ -0,0 +1,125 @@
+<!-- v1.0-->
+<template>
+  <div class="stacked-bar">
+    <div class="wrapper">
+      <div
+        v-for="(segment, index) in computedSegments"
+        :key="index"
+        ref="segmentRefs"
+        :style="{ width: segment.percentage + '%' }"
+        class="segment typo c4-semi-bold"
+        :class="segment.color"
+      >
+        <span class="segment-text">{{ segment.percentage.toFixed(0) }} %</span>
+      </div>
+      <div
+        v-if="availableSegmentValue > 0"
+        :style="{ width: availableSegmentValue + '%' }"
+        class="segment available typo c4-semi-bold"
+      ></div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, onUnmounted, ref } from 'vue'
+import type { Color } from '@core/types/color.type'
+
+type StackedBarDataType = {
```

Comment:
No need to add the `Type` suffix here:
```suggestion
type StackedBarData = {
```